cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/centrality/ClosenessCentrality.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.algorithms.centrality
+
+import com.twitter.cassovary.graph.{GraphDir, BreadthFirstTraverser, Walk, DirectedGraph}
+import scala.collection.immutable
+
+/**
+ * Calculate the closeness centrality of <code>graph</code>
+ * @param normalize Normalize by the ratio of number of connections to other nodes
+ *                  and the number of nodes in the graph
+ */
+class ClosenessCentrality(graph: DirectedGraph, normalize: Boolean = true) extends AbstractCentrality(graph) {
+
+  override def recalculate(): immutable.IndexedSeq[Double] = {
+    graph foreach { node =>
+      val bfs = new BreadthFirstTraverser(graph, GraphDir.OutDir, Seq(node.id), Walk.Limits())
+      var sum = 0.0
+      var neighborCount = 0.0
+
+      bfs foreach { n =>
+        sum += bfs.depth(n.id).get
+        neighborCount += 1
+      }
+
+      centralityValues(node.id) = {
+        if (sum > 0 && graph.nodeCount > 1) {
+          val numerator = neighborCount - 1
+          val denom = if (normalize) (graph.nodeCount - 1) / (neighborCount - 1) else 1
+          numerator / (denom * sum)
```

Comment:
just confirming that it is either (neighborCount - 1) / sum or (neighborCount-1)^2 / ((nodeCount - 1) * sum) ?

You are correct.